core/server/server.js

```diff
@@ -82,6 +84,21 @@ const publicConfigSchema = Joi.object({
     prometheus: {
       enabled: Joi.boolean().required(),
     },
+    influx: {
+      enabled: Joi.boolean().required(),
+      uri: Joi.string()
+        .uri()
+        .when('enabled', { is: true, then: Joi.required() }),
+      timeoutMilliseconds: Joi.number()
+        .integer()
+        .min(1)
+        .when('enabled', { is: true, then: Joi.required() }),
+      intervalSeconds: Joi.number()
+        .integer()
+        .min(1)
+        .when('enabled', { is: true, then: Joi.required() }),
+      hostnameAsAnInstanceId: Joi.boolean(),
```

Comment:
> I think this could be made a little more self-documenting:
> 
> Maybe it could be `instanceIdFrom: Joi.string().equal('hostname', 'heroku-dyno-id', 'random')`.
> 

I agree this is more self-documenting. In this PR wanted to avoid anything related to Heroku in the code. That's why I added the `INSTANCE_ID` env var and proposed to set it in `Procfile` (`INSTANCE_ID=$HEROKU_DYNO_ID`). 
When someone set `instanceIdFrom = heroku-dyno-id` we will read `$HEROKU_DYNO_ID` as an instance id in the code. 

What do you thing about `instanceIdFrom: Joi.string().equal('hostname', 'env-var', 'random')` and a separate option `instanceIdEnvVarName`, which can be set to `HEROKU_DYNO_ID`? I this this is more general solution.

 
> Also while we're still on the current/legacy hosting setup, if there's a way to fetch the public IP for the production server, we could put the mapping somewhere in code. A little gross, but gets the job done.

Why do you want to fetch the public IP? Did you mean [Cnames](https://github.com/badges/shields/blob/master/doc/production-hosting.md#badge-servers)?
Currently we are reading a hostname. We can add a mapping inside our config. 
```
public:
  metrics:
    influx:
      influx: 
        instanceIdFrom: hostname
        hosnameAliases:
          vps71670: s0.servers.shields.io
          vps244529: s1.servers.shields.io
          vps117870: s2.servers.shields.io
```

Reply:
I think it is okay to carefully make references to Heroku in the code. It's not only a platform we are wanting to support for future production. We also support it for staging, review apps, and self-hosting. As long as we're careful to not assume we're on Heroku I don't see why it would be a problem.

That said, I think `instanceIdEnvVarName` in config is a fine compromise.

Reply:
Done. 

> I think it is okay to carefully make references to Heroku in the code. It's not only a platform we are wanting to support for future production. We also support it for staging, review apps, and self-hosting. As long as we're careful to not assume we're on Heroku I don't see why it would be a problem.

I really like your way of thinking.